### PR TITLE
refactor: improve eventarc go samples

### PR DESCRIPTION
**Pull request by grant**
Slightly improves Eventarc samples.

Addresses many comments from: https://github.com/GoogleCloudPlatform/golang-samples/pull/1767 (Thanks for the suggestions)

Feel free to add more suggestions, or 👍  if all good.

### CODE REVIEW

**Comment by grant** [comment]
Can I get a review @tbpg @grayside? Thanks!

File: eventarc/generic/main.go [hunk 1]
```diff
@@ -36,12 +36,15 @@ func GenericHandler(w http.ResponseWriter, r *http.Request) {
 	logAndRespond(w, "Event received!")
 
 	// Log all headers besides authorization header
+	// Assumes headers don't have duplicate keys
```
Comment (tbpg) [inline eventarc/generic/main.go:39]: Why? `strings.Join(v, ",")` can join the values into a single string. Alternatively, keep it as a `[]string`.

File: eventarc/generic/main.go [hunk 2]
```diff
@@ -62,11 +65,10 @@ func GenericHandler(w http.ResponseWriter, r *http.Request) {
 		Headers: headerMap,
 		Body:    body,
 	}
-	out, err := json.Marshal(res)
-	if err != nil {
-		panic(err)
+	if err := json.NewEncoder(w).Encode(res); err != nil {
```
Comment (tbpg) [inline eventarc/generic/main.go:68]: I'm a bit confused -- why write the body and heads above _and_ in JSON here? I might be missing something.

**Review by tbpg — CHANGES_REQUESTED** [review]

File: eventarc/generic/main.go [hunk 3]
```diff
@@ -62,11 +65,10 @@ func GenericHandler(w http.ResponseWriter, r *http.Request) {
...
+		http.Error(w, "Could not marshal JSON output", 500)
```
Comment (tbpg) [inline eventarc/generic/main.go:69]: Should probably `log` the error.

**Review by tbpg — CHANGES_REQUESTED** [review]

File: eventarc/generic/main.go (hunk 1 above)
Comment (grant) [reply eventarc/generic/main.go]: I suppose I haven't seen multiple non-unique headers in a HTTP request seen here.
Changed to the suggestion, thanks.

File: eventarc/generic/main.go (hunk 2 above)
Comment (grant) [reply eventarc/generic/main.go:63]: Thanks for catching. You're right, we already respond here so no need to respond above.

Simplified the sample a bit.

**Review by grant — COMMENTED** [review]
Thanks for the review @tbpg, PTAL.

File: eventarc/generic/main.go (hunk 3 above)
Comment (grant) [reply eventarc/generic/main.go:65]: Now also logs the error.

File: eventarc/generic/main.go [hunk 4]
```diff
@@ -24,34 +24,32 @@ import (
...
+	log.Println("HEADERS:")
 	headerMap := make(map[string]string)
 	for k, v := range r.Header {
-		headerMap[k] = string(v[0])
-		logAndRespond(w, fmt.Sprintf("%q: %q\n", k, v[0]))
+		val := strings.Join(v, ",")
+		if k != "Authorization" {
+			headerMap[k] = string(val)
+			log.Println(fmt.Sprintf("%q: %q\n", k, val))
+		}
```
Comment (tbpg) [inline eventarc/generic/main.go:42]:
```suggestion
		if k != "Authorization" {
			val := strings.Join(v, ",")
			headerMap[k] = val
			log.Println(fmt.Sprintf("%q: %q\n", k, val))
		}
```

You could also make `headerMap` a `map[string][]string` and not have to do `strings.Join`.

File: eventarc/generic/main.go [hunk 5]
```diff
@@ -62,11 +60,11 @@ func GenericHandler(w http.ResponseWriter, r *http.Request) {
 		Headers: headerMap,
 		Body:    body,
 	}
-	out, err := json.Marshal(res)
-	if err != nil {
-		panic(err)
+	if err := json.NewEncoder(w).Encode(res); err != nil {
+		log.Printf("error encoding response: %v", err)
+		http.Error(w, "Could not marshal JSON output", 500)
```
Comment (tbpg) [inline eventarc/generic/main.go:65]:
```suggestion
		http.Error(w, "Could not marshal JSON output", 500)
		return
```

**Review by tbpg — CHANGES_REQUESTED** [review]
A couple small comments.

**Review by tbpg — APPROVED** [review]

**Pull request merged**